extensions/commands/recipe/cmd_create_sbom.py

```diff
@@ -0,0 +1,141 @@
+import os.path
+import sys
+import typing
+
+from conan.api.output import cli_out_write, ConanOutput
+from conan.api.conan_api import ConanAPI
+from conan.cli.args import common_graph_args, validate_common_graph_args
+from conan.cli.command import conan_command
+
+if typing.TYPE_CHECKING:
+    from conans.client.graph.graph import Node
+
+
+def package_type_to_component_type(pt: str):
+    from cyclonedx.model.component import ComponentType
+    if pt is "application":
+        return ComponentType.APPLICATION
+    else:
+        return ComponentType.LIBRARY
+
+
+def licenses(ids):
+    """
+    see https://cyclonedx.org/docs/1.4/json/#components_items_licenses
+    """
+    from cyclonedx.factory.license import LicenseFactory
+    from cyclonedx.model import LicenseChoice
+    if ids is None:
+        return None
+    if not isinstance(ids, tuple):
+        ids = [ids]
+    return [LicenseChoice(license=LicenseFactory().make_from_string(i)) for i in ids]
```

Comment:
I know of reports telling that multiple licenses added in the CycloneDX library caused invalid results.
You should use only one license, if you can.

Reply:
I am not sure how to tackle that. The CycloneDX standard allows multiple licenses. It is supported by the library. The generated JSON is correct and dependencytrack for example has no problem parsing it (according to the log). What should I do? If I pick only one, which one?

Reply:
do you have an example of a Conan package with multiple licenses?

Reply:
bliss has two: https://conan.io/center/recipes/bliss?version=0.77
https://github.com/conan-io/conan-center-index/blob/master/recipes/bliss/all/conanfile.py#L17

Reply:
could you try a project with this `bliss` dependency and upload the CycloneDX result as a gist? 
https://gist.github.com/

i really love to see whether everything results in a valid result for this case.
as i mentioned, there were reports that multi-licenses were broken in the library, but i do not remember seeing any examples of an issue there. (despite the code shows evidence that something might be off)

Reply:
https://gist.github.com/hedtke/f7f2866fbd80504a6d3bae97fc61ad92

Reply:
> https://gist.github.com/hedtke/f7f2866fbd80504a6d3bae97fc61ad92

fantastic. than you so much. relevant part:
https://gist.github.com/hedtke/f7f2866fbd80504a6d3bae97fc61ad92#file-sbom-json-L33-L44

This s just as expected:
- multiple licenses in the document :+1: - so the library has none of the reported issues
- all licenses are names, as they do not match any of the well-known SPCDX licenses,
  according to https://github.com/CycloneDX/specification/blob/master/schema/spdx.schema.json
  compared to https://spdx.org/licenses/

----

**conclusion:** you are golden.